tests/unit_tests/dft/source/workspace_external_tests.cpp

```diff
@@ -0,0 +1,402 @@
+        desc_buf = std::move(desc_buf2);
+        desc_usm.set_value(config_param::WORKSPACE_PLACEMENT, config_value::WORKSPACE_EXTERNAL);
+        desc_buf.set_value(config_param::WORKSPACE_PLACEMENT, config_value::WORKSPACE_EXTERNAL);
+        commit_descriptor(desc_usm, sycl_queue);
+        commit_descriptor(desc_buf, sycl_queue);
+    }
+
+    desc_usm.get_value(config_param::WORKSPACE_EXTERNAL_BYTES, &workspace_bytes);
+    EXPECT_GE(workspace_bytes, 0);
+
+    // We haven't set a workspace, so the following should fail;
+    EXPECT_THROW(compute_forward(desc_usm, fft_data_usm), oneapi::mkl::invalid_argument);
+    sycl_queue.wait_and_throw();
+    EXPECT_THROW(compute_forward(desc_usm, fft_data_buf), oneapi::mkl::invalid_argument);
+    sycl_queue.wait_and_throw();
+
+    if (workspace_bytes > 0) {
+        EXPECT_THROW(desc_usm.set_workspace(nullptr), oneapi::mkl::invalid_argument);
+        sycl::buffer<float> undersize_workspace(
+            static_cast<std::size_t>(workspace_bytes) / sizeof(float) - 1);
+        EXPECT_THROW(desc_buf.set_workspace(undersize_workspace), oneapi::mkl::invalid_argument);
+    }
+
+    usm_workspace = sycl::malloc_device<float>(
+        static_cast<std::size_t>(workspace_bytes) / sizeof(float), sycl_queue);
+    sycl::buffer<float> bufferWorkspace(static_cast<std::size_t>(workspace_bytes) / sizeof(float));
+
+    desc_usm.set_workspace(usm_workspace);
+    desc_buf.set_workspace(bufferWorkspace);
+
+    // Should work:
+    compute_forward(desc_usm, fft_data_usm);
+    sycl_queue.wait_and_throw();
+    compute_forward(desc_buf, fft_data_buf);
+    sycl_queue.wait_and_throw();
+    compute_backward(desc_usm, fft_data_usm);
+    sycl_queue.wait_and_throw();
+    compute_backward(desc_buf, fft_data_buf);
+    sycl_queue.wait_and_throw();
+
+    // Should not work:
+    EXPECT_THROW(compute_forward(desc_usm, fft_data_buf), oneapi::mkl::invalid_argument);
+    EXPECT_THROW(compute_forward(desc_buf, fft_data_usm), oneapi::mkl::invalid_argument);
+    EXPECT_THROW(compute_backward(desc_usm, fft_data_buf), oneapi::mkl::invalid_argument);
+    EXPECT_THROW(compute_backward(desc_buf, fft_data_usm), oneapi::mkl::invalid_argument);
+    sycl_queue.wait_and_throw();
+
+    // Free any allocations:
+    sycl::free(usm_workspace, sycl_queue);
+    sycl::free(fft_data_usm, sycl_queue);
+}
+
+TEST_P(WorkspaceExternalTests, RecommitBehaviour) {
+    using namespace oneapi::mkl::dft;
+    sycl::queue sycl_queue(*GetParam());
+    const int dft_len = 1024 * 3 * 5 * 7 * 16; // A size likely to require an external workspace.
+    float* fft_data_usm = sycl::malloc_device<float>(dft_len * 2, sycl_queue);
+    descriptor<precision::SINGLE, domain::COMPLEX> desc_usm(dft_len);
+    try {
+        // WORKSPACE_EXTERNAL is NOT set.
+        commit_descriptor(desc_usm, sycl_queue);
+    }
+    catch (oneapi::mkl::unimplemented&) {
+        // Admit defeat, and just test that the API does as expected.
```

Comment:
Is there a reason this comment is different from the ones above in a similar situation?
```suggestion
        // DFT size may not be supported. Use a DFT size that probably will be, even if it
        // won't actually use an external workspace internally.
```

Reply:
Applied in [985ea94](https://github.com/oneapi-src/oneMKL/pull/411/commits/985ea9406ed8cbb204e3bdff3b587df010374f1f).